publisher-sdk/src/main/java/com/criteo/publisher/logging/RemoteLogRecordsFactory.kt

```diff
@@ -16,11 +16,86 @@
 
 package com.criteo.publisher.logging
 
+import android.content.Context
+import android.util.Log
+import androidx.annotation.VisibleForTesting
+import com.criteo.publisher.Clock
+import com.criteo.publisher.Session
 import com.criteo.publisher.annotation.OpenForTesting
+import com.criteo.publisher.integration.IntegrationRegistry
+import com.criteo.publisher.logging.RemoteLogRecords.RemoteLogContext
+import com.criteo.publisher.logging.RemoteLogRecords.RemoteLogLevel
+import com.criteo.publisher.logging.RemoteLogRecords.RemoteLogRecord
+import com.criteo.publisher.util.AdvertisingInfo
+import com.criteo.publisher.util.BuildConfigWrapper
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
 
 @OpenForTesting
-internal class RemoteLogRecordsFactory {
+internal class RemoteLogRecordsFactory(
+    private val buildConfigWrapper: BuildConfigWrapper,
+    private val context: Context,
+    private val advertisingInfo: AdvertisingInfo,
+    private val session: Session,
+    private val integrationRegistry: IntegrationRegistry,
+    private val clock: Clock
+) {
+
+  private val iso8601Format = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.ROOT).apply {
+    timeZone = TimeZone.getTimeZone("UTC")
+  }
+
   fun createLogRecords(logMessage: LogMessage): RemoteLogRecords? {
-    TODO("EE-1351")
+    val logLevel = RemoteLogLevel.fromAndroidLogLevel(logMessage.level)
+    val message = createMessageBody(logMessage)
+
+    if (logLevel == null || message == null) {
+      return null
+    }
+
+    val logRecord = RemoteLogRecord(logLevel, listOf(message))
+
+    val context = RemoteLogContext(
+        buildConfigWrapper.sdkVersion,
+        context.packageName,
+        advertisingInfo.advertisingId,
+        session.sessionId,
+        integrationRegistry.profileId,
+        logMessage.throwable?.javaClass?.simpleName
+    )
+
+    return RemoteLogRecords(context, listOf(logRecord))
   }
+
+  @VisibleForTesting
+  fun createMessageBody(logMessage: LogMessage): String? {
+    if (logMessage.message == null && logMessage.throwable == null) {
+      return null
+    }
+
+    val currentDate = Date(clock.currentTimeInMillis)
```

Comment:
nit: move the whole date logic to a Date extension function?

Reply:
This is the only usage of `java.util.Date` in the code base. For now I don't think it is worth extracting such small logic outside.